### ISSUE TITLE
[Bug]: key_press_event not working for TkAgg backend on Windows from Python 3.9.9

### ISSUE THREAD

**Issue by ebardie** [matplotlib/matplotlib#22255]
### Bug summary

On Windows, but not on Linux, with Python 3.9.**8** installed, key_press_events both work for the two examples from the MPL documentation:

[Keypress demo](https://matplotlib.org/stable/gallery/event_handling/keypress_demo.html?highlight=key_press_event)

[Embedding TkInter demo](https://matplotlib.org/stable/gallery/user_interfaces/embedding_in_tk_sgskip.html)

The "Keypress demo" will **not** work with a clean install of Python 3.9.**9** (i.e one not upgraded from an earlier Python 3.9.x release. Deleting a previous install **and** obliterating the install directory is fine. Just upgrading, or uninstalling without obliteration, are not enough.

The problem exists in Python 3.10 on Windows.

The problem is not there for the Qt5Agg backend.

### Code for reproduction

```python
1. Perform a clean install of Python 3.9.9 on Windows.
2. Install matplotlib.
3. Copy the code from the
[Keypress demo](https://matplotlib.org/stable/gallery/event_handling/keypress_demo.html?highlight=key_press_event) in to a file.
4. Run it.
5. Press some keys.
5. Note stdout.
```

### Actual outcome

Stdout will be empty.

### Expected outcome

Stdout should say things like:

press a
press a
press a
press a
press a
press a
press b
press c
press d
press e

### Operating system

Windows

### Matplotlib Version

Any I've tested from 3.3.0 up to 3.5.1

### Matplotlib Backend

TkAgg

### Python version

3.9.9

### Jupyter version

n/a

### Installation

pip

**Comment by ebardie** [comment]
> I'm pretty sure this is a duplicate of #22028. 

Oh, thanks for the headsup @daniilS !


#22028 [occurs after Python 3.10.0](https://github.com/matplotlib/matplotlib/issues/22028#issuecomment-999166163), but #22255  is already present after Python 3.9.8, so they may be related, mainly because of the good news below, but I'm not sure "duplicate" is the right term :)

Also, the following:

>`canvas.get_tk_widget().focus_set()` is a temporary solution 

To #22028 perhaps, but sadly not #22255 :(

I'd previously tried `canvas.get_tk_widget().focus_force()`, but I've just tried `focus_set()` as well, and still no joy :(


However, all that's moot given the good news:

> (I'm hoping to update #22077 soon to fix this properly).

#22077 does "fix" #22255.

I'm not sure that as it stands #22077 is taking the right approach for addressing the problem though:

1. the problem is probably caused by changes lower in the stack, so ideally (I know, I know) would be resolved there too.
2. #22077 changes behaviour, i.e. the user needs to click to set the focus. Ideally, the focus should still be there from the start. 
3. #22255 occurs on Windows but not Linux (I've no Mac to test on), and #22077 affects all platforms.

At the least, don't remove the `focus_set()` call from `__init__()` for unaffected platforms (i.e `if sys.platform != "win32":`), and wrap the `focus_set()` in `button_press_event()` with `if sys.platform == "win32":`.  

Hmmn, this discussion would be better over on #22077.

**Comment by daniilS** [comment]
@ebardie a recent Tk update fixed a bug with focus not being set properly for windows that had not yet been created. It appears Matplotlib depended on the old behaviour. The specific version of Python affected will probably depend on when that it started including Tk 8.6.11 for that platform's distribution. 

#22077 moves `set_focus` to `show()`, which should always work.

I agree that removing the focus call from `__init__()` should be done with a deprecation warning, which is why I still need to edit the PR. However, that only affects embedded plots (which don't use `show()`). But that change, and setting focus on mouse click, is default behaviour for any widget that responds to keyboard input, so shouldn't be made platform-dependent.

**Comment by ebardie** [comment]
@daniilS Cool. Thanks for the update.

**Comment by tacaswell** [comment]
Closed by #22077
